include/swift/Frontend/FrontendOptions.h

```diff
@@ -549,8 +541,18 @@ class FrontendOptions {
     return InputKind == InputFileKind::IFK_Swift &&
            Inputs.hasUniqueInputFilename();
   }
+private:
+  const char *computeSuffix();
+  
+  bool canEmitDependencies() const;
+  bool canEmitHeader() const;
+  bool canEmitLoadedModuleTrace() const;
+  bool canEmitModule() const;
+  
+  /// Return true if changed output filename.
+  bool actionProducesOutputFromFrontend() const;
+  bool actionOutputsToStdout() const;
```

Comment:
This is really "action has textual output".

Reply:
Yes. I renamed it and made it a static function taking an action argument.